core/rovecomm.py

```diff
@@ -73,6 +86,39 @@ def print(self):
         print('----------')
 
 
+class RoveCommCollection:
+    '''
+    Creates a separate thread to read all RoveComm connections
+    '''
+    connections = []
+    shutdown_event = None
+
+    def __init__(self):
+        self.shutdown_event = threading.Event()
+        self.thread = threading.Thread(target=self.listen)
+        self.thread.start()
+
+    def listen(self):
+        while threading.main_thread().is_alive() and not self.shutdown_event.isSet():
+            for connection in self.connections:
+                result = connection.read()
+                if not isinstance(result, list):
+                    result = [result, ]
+                for packet in result:
+                    if packet is not None and packet.data_id != 0:
+                        try:
+                            connection.callbacks[packet.data_id](packet)
+                        except Exception:
+                            pass
+        for connection in self.connections:
+            if isinstance(connection, RoveCommEthernetTcp):
+                connection.close_sockets()
+
```

Comment:
Create addEvent() or something similar, to add a function and data ID to the callbacks dictionary.